### PR TITLE
New analysers for merging recycling container and waste disposal for "Clisson Sèvre et Maine Agglo"

### PR DESCRIPTION
**Pull request by Stefal**
I'm not sure if "item" and "id" numbers are correct for the waste disposal merge analyser.

### CODE REVIEW

**Comment by frodrigo** [comment]
Item should reuse existing item, id should be new.

Please mimic the code layout of other merge analysers.

Add analysers into config file.

Can use original data URL rather than www.stemani.fr?

**Comment by Stefal** [comment]
> 
> 
> Item should reuse existing item, id should be new.

Ok, I've changed ids. But I don't see any already existing item related to waste disposal.
https://github.com/osm-fr/osmose-backend/pull/777/commits/22e8e643677ff5cf23b831eac009c89996ff5c53

> 
> Please mimic the code layout of other merge analysers.

I don't understand. What's wrong with my code ?

> Add analysers into config file.
done https://github.com/osm-fr/osmose-backend/pull/777/commits/61347335740ae646da4c93eb093b5cf970ca1620

> Can use original data URL rather than [www.stemani.fr](http://www.stemani.fr)?
No, the original one is unusable, I had to edit it extensively.

**Comment by frodrigo** [comment]
> Ok, I've changed ids. But I don't see any already existing item related to waste disposal.
Glass recycling item is fine.

>  I don't understand. What's wrong with my code ?
Empty lines, line returns before `)`. Merge code have it own code layout.

> No, the original one is unusable, I had to edit it extensively.
So add it to `/merge_data` directory. Maybe with how to redo the edits.

File: analysers/analyser_merge_recycling_FR_csma.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                    static2 = {"source": self.source},
+                    mapping1 = {
+                               "recycling:glass_bottles": lambda fields: "yes" if fields["detail"] == "Verre" else None,
+                               "recycling:paper": lambda fields: "yes" if fields["detail"] == "Papier" else None,
+                               "recycling:clothes": lambda fields: "yes" if fields["detail"] == "Vêtements" else None,
+                               "location": lambda fields: "underground" if fields["type"] == "Colonne enterrée" else None,
+                               "operator" : lambda fields : "Clisson Sèvre et Maine Agglo" if fields["detail"] != "vêtements"
+                                   else "le Relais",
+                               "opening_hours": lambda fields : "24/7" if "Déchèterie" not in fields["adresse"] and
+                                   "Pôle environnement" not in fields["adresse"] else None,},
```
Comment (frodrigo) [inline analysers/analyser_merge_recycling_FR_csma.py]:
```suggestion
                      "recycling:glass_bottles": lambda fields: "yes" if fields["detail"] == "Verre" else None,
                      "recycling:paper": lambda fields: "yes" if fields["detail"] == "Papier" else None,
                      "recycling:clothes": lambda fields: "yes" if fields["detail"] == "Vêtements" else None,
                      "location": lambda fields: "underground" if fields["type"] == "Colonne enterrée" else None,
                      "operator" : lambda fields: "Clisson Sèvre et Maine Agglo" if fields["detail"] != "vêtements" else "le Relais",
                      "opening_hours": lambda fields : "24/7" if "Déchèterie" not in fields["adresse"] and "Pôle environnement" not in fields["adresse"] else None,},
```

File: analysers/analyser_merge_waste_disposal_FR_csma.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                        "amenity": "waste_disposal",
+                        "operator": "Clisson Sèvre et Maine Agglo",
+                        "fee": "yes",
+                        "authentication:membership_card" : "yes",
+                        "authentication:none" : "no",},
+                    static2 = {"source": self.source},
+                    mapping1 = {
+                               "location": lambda fields: "underground" if fields["type"] == "Colonne enterrée" else None,
+                               "opening_hours": lambda fields : "24/7" if "Déchèterie" not in fields["adresse"] and
+                                   "Pôle environnement" not in fields["adresse"] else None},
```
Comment (frodrigo) [inline analysers/analyser_merge_waste_disposal_FR_csma.py]:
```suggestion
                    "location": lambda fields: "underground" if fields["type"] == "Colonne enterrée" else None,
                    "opening_hours": lambda fields : "24/7" if "Déchèterie" not in fields["adresse"] and "Pôle environnement" not in fields["adresse"] else None},
```

**Comment by Stefal** [comment]
That should be ok now

**Comment by frodrigo** [comment]
Have you ran it successfully?

**Comment by Stefal** [comment]
> 
> 
> Have you ran it successfully?

Yes, but I failed to visualize the result on the frontend. I've only check the xml file.

**Comment by frodrigo** [comment]
OK. Fine. Thank you.

**Comment by frodrigo** [comment]
http://osmose.openstreetmap.fr/en/errors/?item=8120,8121,8122

**Comment by frodrigo** [comment]
Ok, we got the data, please review the results.

**Comment by Stefal** [comment]
I made a mistake in the recycling analyser: I forgot to exclude the waste disposal nodes.

I will create a new pull request.

**Comment by Stefal** [comment]
fix with #782

**Pull request merged**